src/kem/kyber/serialize.rs

```diff
@@ -1,35 +1,28 @@
 use super::{
-    arithmetic::{to_unsigned_representative, KyberFieldElement, KyberPolynomialRingElement},
-    compress::{compress_q, decompress_q},
-    constants::{BYTES_PER_RING_ELEMENT, COEFFICIENTS_IN_RING_ELEMENT, SHARED_SECRET_SIZE},
+    arithmetic::{to_unsigned_representative, FieldElement, PolynomialRingElement},
+    compress::{
+        compress_ciphertext_coefficient, compress_message_coefficient,
+        decompress_ciphertext_coefficient, decompress_message_coefficient,
+    },
+    constants::{BYTES_PER_RING_ELEMENT, SHARED_SECRET_SIZE},
 };
 
+#[cfg(not(hax))]
+use super::constants::COEFFICIENTS_IN_RING_ELEMENT;
+
 #[inline(always)]
 pub(super) fn compress_then_serialize_message(
-    re: KyberPolynomialRingElement,
+    re: PolynomialRingElement,
 ) -> [u8; SHARED_SECRET_SIZE] {
     let mut serialized = [0u8; SHARED_SECRET_SIZE];
 
     for (i, coefficients) in re.coefficients.chunks_exact(8).enumerate() {
         for (j, coefficient) in coefficients.iter().enumerate() {
             let coefficient = to_unsigned_representative(*coefficient);
 
-            let coefficient_compressed = compress_q(1, coefficient);
-
-            // At this point, the following should hold for |coefficient_compressed|:
```

Comment:
Does this comment no longer apply?

Reply:
It does, it's just that it's now enforced by the way `compress_message_coefficient` is written.